pkg/api/types.go

```diff
@@ -2700,6 +2725,10 @@ type ServiceSpec struct {
 	// +optional
 	SessionAffinity ServiceAffinity
 
+	// ServiceAffinityConfig contains the configurations of session affinity.
+	// +optional
+	ServiceAffinityConfig *ServiceAffinityConfig
```

Comment:
We should think about this from the API point of view.  Do we need 2 levels of nesting?  Or should this be embedded.

e.g.

```
sessionAffinity: ClientIP
serviceAffinityConfig:
  clientIPConfig:
    timeoutSeconds: 60
```

vs

```
sessionAffinity: ClientIP
clientIPAffinityConfig:
  timeoutSeconds: 60
```